packages/web-client/app/components/card-pay/dashboard-panel/index.gts

```diff
@@ -0,0 +1,71 @@
+import Component from '@glimmer/component';
+import cn from '@cardstack/boxel/helpers/cn';
+import or from 'ember-truth-helpers/helpers/or';
+import cssUrl from "@cardstack/boxel/helpers/css-url";
+import { svgJar } from '@cardstack/boxel/utils/svg-jar';
+import Panel from './panel';
+
+interface Signature {
+  Element: HTMLDetailsElement;
+  Args: {
+    noCollapse?: boolean;
+    noBottomPadding?: boolean;
+    open?: Function;
+    panel: Panel
+  }
+  Blocks: {
+    detail: []
+    footer: []
+  }
+}
+
+export default class CardPayDashboardPanel extends Component<Signature> {
+  <template>
+    <details
+      class={{cn
+        "dashboard-details"
+        dashboard-details--no-collapse=@noCollapse
+        dashboard-details--no-bottom-padding=@noBottomPadding
+        dashboard-details--has-footer=(has-block 'footer')
+      }}
+      ...attributes
+      {{! @glint-ignore }}
+      open={{or @noCollapse @open}}
```

Comment:
To clarify (and I think @Aierie already meant this but it took me a while to parse it correctly in my head): the `open` attribute on native HTML `<dialog>` already accepts a boolean. The question is about whether the `@open` argument to this component should change from `Function` to `@boolean`.

I think this is a place where we've uncovered a preexisting bug. If anybody does pass a function for `@open` it will do the wrong thing and try to `toString()` the function into the value of the `open` attribute. 

If we don't want to change the call signature, the fix is:

```ts
const asBool = (value: any): boolean => Boolean(value);
<template>
  <details open={{or @noCollapse (asBool @open)}} />
</template>
```

Tangentially, it *should* be possible to just say `open={{or @noCollapse (Boolean @open)}}` since `Boolean` is a function and all functions can be helpers, but typescript and/or glint doesn't like that because of the way `Boolean` is also a constructor.

Reply:
Sorry, I meant that `@open` is actually meant to be a `boolean` argument, but is incorrectly typed as `Function` here, maybe because we often use `@is<adjective>` to refer to states, rather than `@<adjective>`.

Reply:
Elegant resolution to this: `@open` was not being used by anything!